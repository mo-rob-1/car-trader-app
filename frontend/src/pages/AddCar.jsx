import { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import { toast } from "react-toastify";
import { addCar, reset } from "../features/cars/carSlice";

function AddCar() {
  const { user } = useSelector((state) => state.auth);
  const { isLoading, isError, isSuccess, message } = useSelector((state) => state.car);

  const [name] = useState(user.name);
  const [email] = useState(user.email);
  const [model, setModel] = useState("");
  const [price, setPrice] = useState(0);
  const [doors, setDoors] = useState("");
  const [seats, setSeats] = useState("");
  const [imageURL, setImageURL] = useState("");
  const [description, setDescription] = useState("");
  const [phoneNumber, setPhoneNumber] = useState("");
  const [location, setLocation] = useState("");
  const [region, setRegion] = useState("");
  const [make, setMake] = useState("");
  const [fuelType, setFuelType] = useState("");
  const [bodyType, setBodyType] = useState("");
  const [transmission, setTransmission] = useState("");
  const [mileage, setMileage] = useState(0);
  const [colour, setColour] = useState("");
  const [engineSize, setEngineSize] = useState(1.0);

  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    if (isError) {
      toast.error(message);
    }

    if (isSuccess) {
      dispatch(reset());
      navigate("/");
    }

    dispatch(reset());
  }, [dispatch, isError, isSuccess, navigate, message]);

  const onSubmit = (e) => {
    e.preventDefault();
    dispatch(
      addCar({
        make,
        model,
        description,
        price,
        imageURL,
        location,
        region,
        phoneNumber,
        email,
        fuelType,
        transmission,
        doors,
        seats,
        bodyType,
        mileage,
        colour,
        engineSize,
      })
    );
  };

  if (isLoading) {
    return <h3>Loading...</h3>;
  }

  return (
    <>
      <section>
        <h1 className="text-3xl font-bold">Add a car</h1>
      </section>
      <section>
        <form onSubmit={onSubmit}>
          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Seller's Name:</span>
            </label>
            <input className="input input-bordered w-full max-w-xs" id="name" value={name} name="name" disabled />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Seller's Email Address:</span>
            </label>
            <input className="input input-bordered w-full max-w-xs" id="email" value={email} name="email" disabled />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Seller's Phone Number:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              id="phoneNumber"
              name="phoneNumber"
              value={phoneNumber}
              onChange={(e) => setPhoneNumber(e.target.value)}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Make:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              id="make"
              name="make"
              onChange={(e) => setMake(e.target.value)}
              value={make}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Model:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              id="model"
              name="model"
              onChange={(e) => setModel(e.target.value)}
              value={model}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Image URL:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              id="imageURL"
              name="imageURL"
              value={imageURL}
              onChange={(e) => setImageURL(e.target.value)}
              required
            />
          </div>

          <div className="form-control">
            <label className="label">
              <span className="label-text">Description</span>
            </label>
            <textarea
              className="textarea textarea-bordered h-24"
              value={description}
              onChange={(e) => setDescription(e.target.value)}
            ></textarea>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Fuel Type:</span>
            </label>
            <select
              className="select select-bordered"
              id="fuelType"
              name="fuelType"
              value={fuelType}
              onChange={(e) => setFuelType(e.target.value)}
              required
            >
              <option value="" disabled>
                Add fuel type
              </option>
              <option value="Petrol">Petrol</option>
              <option value="Diesel">Diesel</option>
              <option value="Electric">Electric</option>
              <option value="Hybrid">Hybrid</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Engine Size:</span>
            </label>
            <select
              className="select select-bordered"
              id="engineSize"
              name="engineSize"
              value={engineSize}
              onChange={(e) => setEngineSize(e.target.value)}
              required
            >
              <option value="" disabled>
                Select the engine size
              </option>
              <option value="1">1.0 litre</option>
              <option value="1.2">1.2 litres</option>
              <option value="1.4">1.4 litres</option>
              <option value="1.6">1.6 litres</option>
              <option value="1.8">1.8 litres</option>
              <option value="2">2.0 litres</option>
              <option value="2.2">2.2 litres</option>
              <option value="2.4">2.4 litres</option>
              <option value="2.6">2.6 litres</option>
              <option value="2.8">2.8 litres</option>
              <option value="3">3.0 litres</option>
              <option value="4">4.0 litres</option>
              <option value="5">5.0 litres</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Body Type:</span>
            </label>
            <select
              className="select select-bordered"
              id="bodyType"
              name="bodyType"
              value={bodyType}
              onChange={(e) => setBodyType(e.target.value)}
              required
            >
              <option value="" disabled>
                Add body type
              </option>
              <option value="Sedan">Sedan</option>
              <option value="Hatchback">Hatchback</option>
              <option value="SUV">SUV</option>
              <option value="MPV">MPV</option>
              <option value="Coupe">Coupe</option>
              <option value="Saloon">Saloon</option>
              <option value="Estate">Estate</option>
              <option value="Convertible">Convertible</option>
              <option value="Supercar">Supercar</option>
              <option value="Sports">Sports</option>
              <option value="Luxury">Luxury</option>
              <option value="Pickup">Pickup</option>
              <option value="Van">Van</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Transmission:</span>
            </label>
            <select
              className="select select-bordered"
              id="transmission"
              name="transmission"
              value={transmission}
              onChange={(e) => setTransmission(e.target.value)}
              required
            >
              <option value="" disabled>
                Add transmission
              </option>
              <option value="Automatic">Automatic</option>
              <option value="Manual">Manual</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Colour:</span>
            </label>
            <select
              className="select select-bordered"
              id="colour"
              name="colour"
              value={colour}
              onChange={(e) => setColour(e.target.value)}
              required
            >
              <option value="" disabled>
                Select colour
              </option>
              <option value="Black">Black</option>
              <option value="White">White</option>
              <option value="Red">Red</option>
              <option value="Blue">Blue</option>
              <option value="Green">Green</option>
              <option value="Yellow">Yellow</option>
              <option value="Purple">Purple</option>
              <option value="Orange">Orange</option>
              <option value="Pink">Pink</option>
              <option value="Brown">Brown</option>
              <option value="Grey">Grey</option>
              <option value="Silver">Silver</option>
              <option value="Gold">Gold</option>
              <option value="Bronze">Bronze</option>
              <option value="Multicolour">Multicolour</option>
              <option value="Platinum">Platinum</option>
              <option value="Other">Other</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Number of doors:</span>
            </label>
            <select
              className="select select-bordered"
              id="doors"
              name="doors"
              value={doors}
              onChange={(e) => setDoors(e.target.value)}
              required
            >
              <option value="" disabled>
                Select number of doors
              </option>
              <option value="2">2</option>
              <option value="4">4</option>
              <option value="5">5</option>
              <option value="6">6</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Number of seats:</span>
            </label>
            <select
              className="select select-bordered"
              id="seats"
              name="seats"
              value={seats}
              onChange={(e) => setSeats(e.target.value)}
              required
            >
              <option value="" disabled>
                Select number of seats
              </option>
              <option value="2">2</option>
              <option value="4">4</option>
              <option value="5">5</option>
              <option value="6">6</option>
              <option value="8">8</option>
            </select>
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Mileage:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              type="number"
              step="1"
              id="mileage"
              name="mileage"
              value={mileage}
              onChange={(e) => setMileage(e.target.value)}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Price:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              type="number"
              step="0.01"
              id="price"
              name="price"
              value={price}
              onChange={(e) => setPrice(e.target.value)}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Location:</span>
            </label>
            <input
              className="input input-bordered w-full max-w-xs"
              id="location"
              name="location"
              value={location}
              onChange={(e) => setLocation(e.target.value)}
              required
            />
          </div>

          <div className="form-control w-full max-w-xs">
            <label className="label">
              <span className="label-text">Pick a region:</span>
            </label>
            <select
              className="select select-bordered"
              id="region"
              name="region"
              value={region}
              onChange={(e) => setRegion(e.target.value)}
              required
            >
              <option value="" disabled>
                Add a region
              </option>
              <option value="London">London</option>
              <option value="North West">North West</option>
              <option value="North East">North East</option>
              <option value="Yorkshire">Yorkshire</option>
              <option value="Midlands">Midlands</option>
              <option value="South East">South East</option>
              <option value="South West">South West</option>
              <option value="Wales">Wales</option>
              <option value="Scotland">Scotland</option>
            </select>
          </div>

          <div>
            <button className="btn">Submit</button>
          </div>
        </form>
      </section>
    </>
  );
}

export default AddCar;
