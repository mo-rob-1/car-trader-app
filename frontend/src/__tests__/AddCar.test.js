import { render, screen, within } from "@testing-library/react";
import AddCar from "../pages/AddCar";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";

describe("AddCar", () => {
  it("renders the add car page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the seller's contact number input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const contactNumberLabel = getByTestId("contact-number-label");
    expect(contactNumberLabel).toBeTruthy();

    const contactNumberInput = getByTestId("contact-number-input");
    expect(contactNumberInput).toBeTruthy();
  });

  it("renders the make input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const makeLabel = getByTestId("make-label");
    expect(makeLabel).toBeTruthy();

    const makeInput = getByTestId("make-input");
    expect(makeInput).toBeTruthy();
  });

  it("renders the model input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const modelLabel = getByTestId("model-label");
    expect(modelLabel).toBeTruthy();

    const modelInput = getByTestId("model-input");
    expect(modelInput).toBeTruthy();
  });

  it("renders the image input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const imageLabel = getByTestId("image-label");
    expect(imageLabel).toBeTruthy();

    const imageInput = getByTestId("image-input");
    expect(imageInput).toBeTruthy();
  });

  it("renders the description input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const descriptionLabel = getByTestId("desc-label");
    expect(descriptionLabel).toBeTruthy();

    const descriptionInput = getByTestId("desc-textarea");
    expect(descriptionInput).toBeTruthy();
  });

  it("renders the location input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const locationLabel = getByTestId("location-label");
    expect(locationLabel).toBeTruthy();

    const locationInput = getByTestId("location-input");
    expect(locationInput).toBeTruthy();
  });

  it("renders the region select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const regionLabel = getByTestId("region-label");
    expect(regionLabel).toBeTruthy();

    const regionSelect = getByTestId("region-select");
    expect(regionSelect).toBeTruthy();
  });

  it("renders the mileage input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const mileageLabel = getByTestId("mileage-label");
    expect(mileageLabel).toBeTruthy();

    const mileageInput = getByTestId("mileage-input");
    expect(mileageInput).toBeTruthy();
  });

  it("renders the fuel type select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const fuelTypeLabel = getByTestId("fuel-type-label");
    expect(fuelTypeLabel).toBeTruthy();

    const fuelTypeSelect = getByTestId("fuel-type-select");
    expect(fuelTypeSelect).toBeTruthy();
  });

  it("renders the engine size select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const engineSizeLabel = getByTestId("engine-size-label");
    expect(engineSizeLabel).toBeTruthy();

    const engineSizeSelect = getByTestId("engine-size-select");
    expect(engineSizeSelect).toBeTruthy();
  });

  it("renders the body type select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const bodyTypeLabel = getByTestId("body-type-label");
    expect(bodyTypeLabel).toBeTruthy();

    const bodyTypeSelect = getByTestId("body-type-select");
    expect(bodyTypeSelect).toBeTruthy();
  });

  it("renders the transmission select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const transmissionLabel = getByTestId("transmission-label");
    expect(transmissionLabel).toBeTruthy();

    const transmissionSelect = getByTestId("transmission-select");
    expect(transmissionSelect).toBeTruthy();
  });

  it("renders the color select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const colorLabel = getByTestId("colour-label");
    expect(colorLabel).toBeTruthy();

    const colorSelect = getByTestId("colour-select");
    expect(colorSelect).toBeTruthy();
  });

  it("renders the number of doors select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const numberOfDoorsLabel = getByTestId("number-of-doors-label");
    expect(numberOfDoorsLabel).toBeTruthy();

    const numberOfDoorsSelect = getByTestId("number-of-doors-select");
    expect(numberOfDoorsSelect).toBeTruthy();
  });

  it("renders the number of seats select and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const numberOfSeatsLabel = getByTestId("number-of-seats-label");
    expect(numberOfSeatsLabel).toBeTruthy();

    const numberOfSeatsSelect = getByTestId("number-of-seats-select");
    expect(numberOfSeatsSelect).toBeTruthy();
  });

  it("renders the price input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const priceLabel = getByTestId("price-label");
    expect(priceLabel).toBeTruthy();

    const priceInput = getByTestId("price-input");
    expect(priceInput).toBeTruthy();
  });

  it("renders the submit button", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const submitButton = getByTestId("submit-button");
    expect(submitButton).toBeTruthy();
  });

  it("renders the sellers name input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const sellersNameLabel = getByTestId("sellers-name-label");
    expect(sellersNameLabel).toBeTruthy();

    const sellersNameInput = getByTestId("sellers-name-input");
    expect(sellersNameInput).toBeTruthy();
  });

  it("renders the sellers email input and label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    const sellersEmailLabel = getByTestId("sellers-email-label");
    expect(sellersEmailLabel).toBeTruthy();

    const sellersEmailInput = getByTestId("sellers-email-input");
    expect(sellersEmailInput).toBeTruthy();
  });

  it("renders the Add Car text heading", () => {
    render(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>
    );
    screen.getByRole("heading", {
      name: /add a car/i,
    });
  });
});
