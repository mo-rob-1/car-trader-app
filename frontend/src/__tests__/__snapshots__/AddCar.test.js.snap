// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCar renders the add car page 1`] = `
<DocumentFragment>
  <div
    class="bg-base-200"
  >
    <div
      class="m-auto w-11/12 max-w-screen-2xl pt-5 pb-5 md:pt-12 md:pb-12"
    >
      <section
        class="mt-6 bg-white p-5 md:p-8 rounded-2xl"
      >
        <section
          class="mb-6"
        >
          <h1
            class="text-3xl font-bold"
          >
            Add a car
          </h1>
        </section>
        <form
          class="grid gap-3"
        >
          <div
            class="md:flex"
          >
            <div
              class="form-control w-full max-w-xs md:mr-5"
            >
              <label
                class="label"
              >
                <span
                  class="label-text"
                  data-testid="sellers-name-label"
                >
                  Seller's Name:
                </span>
              </label>
              <input
                class="input input-bordered w-full max-w-xs !bg-white"
                data-testid="sellers-name-input"
                disabled=""
                id="name"
                name="name"
                value=""
              />
            </div>
            <div
              class="form-control w-full max-w-xs"
            >
              <label
                class="label"
              >
                <span
                  class="label-text"
                  data-testid="sellers-email-label"
                >
                  Seller's Email Address:
                </span>
              </label>
              <input
                class="input input-bordered w-full max-w-xs !bg-white"
                data-testid="sellers-email-input"
                disabled=""
                id="email"
                name="email"
                value=""
              />
            </div>
          </div>
          <hr
            class="mt-4"
          />
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="contact-number-label"
              >
                Seller's Contact Number:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="contact-number-input"
              id="phoneNumber"
              name="phoneNumber"
              required=""
              value=""
            />
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="make-label"
              >
                Make:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="make-input"
              id="make"
              name="make"
              required=""
              value=""
            />
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="model-label"
              >
                Model:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="model-input"
              id="model"
              name="model"
              required=""
              value=""
            />
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="image-label"
              >
                Image URL:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="image-input"
              id="imageURL"
              name="imageURL"
              required=""
              value=""
            />
            <label
              class="label"
            >
              <span
                class="label-text-alt"
              >
                E.g. https://www.domain.com/ford-focus.jpeg
              </span>
            </label>
          </div>
          <div
            class="form-control"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="desc-label"
              >
                Description:
              </span>
            </label>
            <textarea
              class="textarea textarea-bordered h-24 lg:w-3/5 xl:w-2/5"
              data-testid="desc-textarea"
              maxlength="2000"
            />
            <label
              class="label"
            >
              <span
                class="label-text-alt"
              >
                Max 2000 characters - 0 / 2000
              </span>
            </label>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="location-label"
              >
                Location:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="location-input"
              id="location"
              name="location"
              required=""
              value=""
            />
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="region-label"
              >
                Region:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="region-select"
              id="region"
              name="region"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select region
              </option>
              <option
                value="London"
              >
                London
              </option>
              <option
                value="North West"
              >
                North West
              </option>
              <option
                value="North East"
              >
                North East
              </option>
              <option
                value="Yorkshire"
              >
                Yorkshire
              </option>
              <option
                value="Midlands"
              >
                Midlands
              </option>
              <option
                value="South East"
              >
                South East
              </option>
              <option
                value="South West"
              >
                South West
              </option>
              <option
                value="Wales"
              >
                Wales
              </option>
              <option
                value="Scotland"
              >
                Scotland
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="mileage-label"
              >
                Mileage:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="mileage-input"
              id="mileage"
              name="mileage"
              required=""
              step="1"
              type="number"
              value="0"
            />
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="fuel-type-label"
              >
                Fuel Type:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="fuel-type-select"
              id="fuelType"
              name="fuelType"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select fuel type
              </option>
              <option
                value="Petrol"
              >
                Petrol
              </option>
              <option
                value="Diesel"
              >
                Diesel
              </option>
              <option
                value="Hybrid"
              >
                Hybrid
              </option>
              <option
                value="Electric"
              >
                Electric
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="engine-size-label"
              >
                Engine Size:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="engine-size-select"
              id="engineSize"
              name="engineSize"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select the engine size
              </option>
              <option
                value="1"
              >
                1 litres
              </option>
              <option
                value="1.2"
              >
                1.2 litres
              </option>
              <option
                value="1.4"
              >
                1.4 litres
              </option>
              <option
                value="1.6"
              >
                1.6 litres
              </option>
              <option
                value="1.8"
              >
                1.8 litres
              </option>
              <option
                value="2"
              >
                2 litres
              </option>
              <option
                value="2.2"
              >
                2.2 litres
              </option>
              <option
                value="2.4"
              >
                2.4 litres
              </option>
              <option
                value="2.6"
              >
                2.6 litres
              </option>
              <option
                value="2.8"
              >
                2.8 litres
              </option>
              <option
                value="3"
              >
                3 litres
              </option>
              <option
                value="4"
              >
                4 litres
              </option>
              <option
                value="5"
              >
                5 litres
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="body-type-label"
              >
                Body Type:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="body-type-select"
              id="bodyType"
              name="bodyType"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select body type
              </option>
              <option
                value="Sedan"
              >
                Sedan
              </option>
              <option
                value="Hatchback"
              >
                Hatchback
              </option>
              <option
                value="SUV"
              >
                SUV
              </option>
              <option
                value="MPV"
              >
                MPV
              </option>
              <option
                value="Coupe"
              >
                Coupe
              </option>
              <option
                value="Saloon"
              >
                Saloon
              </option>
              <option
                value="Estate"
              >
                Estate
              </option>
              <option
                value="Convertible"
              >
                Convertible
              </option>
              <option
                value="Supercar"
              >
                Supercar
              </option>
              <option
                value="Sports"
              >
                Sports
              </option>
              <option
                value="Luxury"
              >
                Luxury
              </option>
              <option
                value="Pickup"
              >
                Pickup
              </option>
              <option
                value="Van"
              >
                Van
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="transmission-label"
              >
                Transmission:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="transmission-select"
              id="transmission"
              name="transmission"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select transmission
              </option>
              <option
                value="Automatic"
              >
                Automatic
              </option>
              <option
                value="Manual"
              >
                Manual
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="colour-label"
              >
                Colour:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="colour-select"
              id="colour"
              name="colour"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select colour
              </option>
              <option
                value="Black"
              >
                Black
              </option>
              <option
                value="White"
              >
                White
              </option>
              <option
                value="Red"
              >
                Red
              </option>
              <option
                value="Blue"
              >
                Blue
              </option>
              <option
                value="Green"
              >
                Green
              </option>
              <option
                value="Yellow"
              >
                Yellow
              </option>
              <option
                value="Purple"
              >
                Purple
              </option>
              <option
                value="Orange"
              >
                Orange
              </option>
              <option
                value="Pink"
              >
                Pink
              </option>
              <option
                value="Brown"
              >
                Brown
              </option>
              <option
                value="Grey"
              >
                Grey
              </option>
              <option
                value="Silver"
              >
                Silver
              </option>
              <option
                value="Gold"
              >
                Gold
              </option>
              <option
                value="Bronze"
              >
                Bronze
              </option>
              <option
                value="Multicolour"
              >
                Multicolour
              </option>
              <option
                value="Platinum"
              >
                Platinum
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="number-of-doors-label"
              >
                Number of doors:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="number-of-doors-select"
              id="doors"
              name="doors"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select number of doors
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="number-of-seats-label"
              >
                Number of seats:
              </span>
            </label>
            <select
              class="select select-bordered"
              data-testid="number-of-seats-select"
              id="seats"
              name="seats"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Select number of seats
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
              <option
                value="8"
              >
                8
              </option>
            </select>
          </div>
          <div
            class="form-control w-full max-w-xs"
          >
            <label
              class="label"
            >
              <span
                class="label-text"
                data-testid="price-label"
              >
                Price:
              </span>
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="price-input"
              id="price"
              name="price"
              required=""
              step="0.01"
              type="number"
              value="0"
            />
          </div>
          <div
            class="mt-3"
          >
            <button
              class="btn"
              data-testid="submit-button"
            >
              Submit
            </button>
          </div>
        </form>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
